### ISSUE TITLE
Add WASM shim

### ISSUE THREAD

**Issue by k82cn** [xflops/flame#72]
[wasmtime](https://github.com/bytecodealliance/wasmtime/) is a WASM runtime, it's easy to integrate with high performance.

**Comment by k82cn** [comment]
Add service interface:

```
trait FlameService {
  on_service_create();
  on_session_enter();
  on_task_invoke();
  on_session_leave();
  on_service_destroy();
}
```

Add WASM wrapper to get/set parameters of them, and the WASM should start with `flame:run`, for example:

```
struct MySvc {
};

impl FlameService for MySvc {
}

fn main() {
    let my_svc = xxx;
    ...
   flame::run(my_svc);
}
```

**Comment by k82cn** [comment]
fixed by #78 #77 #75
